src/orion/benchmark/__init__.py

```diff
@@ -0,0 +1,330 @@
+        self._id = None
+        self.name = name
+        self.algorithms = algorithms
+        self.targets = targets
+        self.metadata = {}
+
+        self.studies = []
+
+    def setup_studies(self):
+        """Setup studies to run for the benchmark.
+        Benchmark `algorithms`, together with each `task` and `assessment` combination
+        define a study.
+        """
+        for target in self.targets:
+            assessments = target["assess"]
+            tasks = target["task"]
+
+            for assess, task in itertools.product(*[assessments, tasks]):
+                study = Study(self, self.algorithms, assess, task)
+                study.setup_experiments()
+                self.studies.append(study)
+
+    def process(self):
+        """Run studies experiment"""
+        for study in self.studies:
+            study.execute()
+
+    def status(self, silent=True):
+        """Display benchmark status"""
+        total_exp_num = 0
+        complete_exp_num = 0
+        total_trials = 0
+        benchmark_status = []
+        for study in self.studies:
+            for status in study.status():
+                column = dict()
+                column["Algorithms"] = status["algorithm"]
+                column["Assessments"] = status["assessment"]
+                column["Tasks"] = status["task"]
+                column["Total Experiments"] = status["experiments"]
+                total_exp_num += status["experiments"]
+                column["Completed Experiments"] = status["completed"]
+                complete_exp_num += status["completed"]
+                column["Submitted Trials"] = status["trials"]
+                total_trials += status["trials"]
+                benchmark_status.append(column)
+
+        if not silent:
+            print(
+                "Benchmark name: {}, Experiments: {}/{}, Submitted trials: {}".format(
+                    self.name, complete_exp_num, total_exp_num, total_trials
+                )
+            )
+            try:
+                from IPython.display import HTML, display
+
+                display(
+                    HTML(
+                        tabulate(
+                            benchmark_status,
+                            headers="keys",
+                            tablefmt="html",
+                            stralign="center",
+                            numalign="center",
+                        )
+                    )
+                )
+            except ImportError:
+                table = tabulate(
+                    benchmark_status,
+                    headers="keys",
+                    tablefmt="grid",
+                    stralign="center",
+                    numalign="center",
+                )
+                print(table)
+
+        return benchmark_status
+
+    def analysis(self):
+        """Return all the assessment figures"""
+        figures = []
+        for study in self.studies:
+            figure = study.display()
```

Comment:
The names are a bit confusing I think. Experiment has analysis, study has display and assessment has plot_figures. As I understand it they all are providing figures for the analysis? I would reuse the same name across the different class.

Reply:
Good idea. I was trying to differentiate at the beginning as they are display at different levels, but seems make it more confusing.